src/environment.rs

```diff
@@ -145,7 +147,10 @@ impl Environment {
 
     /// Create a read-write transaction for use with the environment. This method will block while
     /// there are any other read-write transactions open on the environment.
-    pub fn begin_rw_txn<'env>(&'env self) -> Result<RwTransaction<'env>> {
+    pub fn begin_rw_txn<'env>(&'env self, headroom: Option<usize>) -> Result<RwTransaction<'env>> {
+        while self.needs_resize(headroom)? {
+            self.do_resize(headroom.unwrap_or(1 << 28))?;
+        }
         RwTransaction::new(self)
```

Comment:
What happens when two rw transactions are started simultaneously? Say there is enough space for each but not for both. Each of them separately checks there is enough space, and `RwTransaction::new` is started twice, but there is not enough space for both to complete.

Reply:
Good point. I guess we'll have to introduce some kind of locking or an atomic_flag with a spin lock. Lmdb prevents concurrent transactions anyway.

Reply:
I don't know if the last comment I wrote got lost, but I added a mutex to fix this.